pkg/user/registry/user/rest.go

```diff
@@ -1,44 +1,71 @@
 package user
 
 import (
-	"errors"
+	"fmt"
 
 	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
 	kerrs "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
-	"github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
 
 	"github.com/openshift/origin/pkg/user/api"
 	"github.com/openshift/origin/pkg/user/api/validation"
 )
 
-// REST implements the RESTStorage interface in terms of an Registry.
-type REST struct {
-	registry Registry
+// userStrategy implements behavior for Users
+type userStrategy struct {
+	runtime.ObjectTyper
 }
 
-// NewREST returns a new REST.
-func NewREST(registry Registry) apiserver.RESTStorage {
-	return &REST{registry}
+// Strategy is the default logic that applies when creating and updating User
+// objects via the REST API.
+var Strategy = userStrategy{kapi.Scheme}
+
+// NamespaceScoped is false for users
+func (userStrategy) NamespaceScoped() bool {
+	return false
+}
+
+func (userStrategy) GenerateName(base string) string {
+	return base
+}
+
+func (userStrategy) ResetBeforeCreate(obj runtime.Object) {
+}
+
+// Validate validates a new user
+func (userStrategy) Validate(obj runtime.Object) kerrs.ValidationErrorList {
+	user := obj.(*api.User)
```

Comment:
copied from other Validate methods... a cast without a double-assignment to a boolean panics if it fails.